fe/src/main/java/org/apache/doris/qe/SqlModeHelper.java

```diff
@@ -139,42 +143,75 @@ public static String decode(Long sqlMode) throws AnalysisException {
     }
 
     // convert string type SQL MODE to long type that session can store
-    public static Long encode(String sqlMode) throws AnalysisException {
+    public static Long encode(String sqlMode) throws DdlException {
         List<String> names =
                 Splitter.on(',').trimResults().omitEmptyStrings().splitToList(sqlMode);
 
         // empty string parse to 0
         long value = 0L;
         for (String key : names) {
-            // the SQL MODE must be supported, set sql mode repeatedly is not allowed
-            if (!isSupportedSqlMode(key) || (value & getSupportedSqlMode().get(key)) != 0) {
-                throw new AnalysisException("Variable 'sql_mode' can't be set to the value of: " + key);
+            SqlModeI sqlModeI;
```

Comment:
I think you make it more complicate
I think some functions is enough

1. expand combine code
2. get code from string.